src/SlmLocale/Strategy/UriPathStrategy.php

```diff
@@ -240,12 +277,21 @@ protected function getAliasForLocale($locale)
         }
     }
 
-    protected function getBasePath()
+    /**
+     * @param RequestInterface|null $request
+     * @return string|null
+     */
+    protected function getBasePath(RequestInterface $request = null)
     {
+        $result = null;
         if ($this->router instanceof TreeRouteStack) {
-            return $this->router->getBaseUrl();
+            $result = $this->router->getBaseUrl();
+        }
+
+        if (null === $result && null !== $request && method_exists($request, 'getBasePath')) {
```

Comment:
why router does not have baseUrl?

Reply:
check **https://github.com/basz/SlmLocale/issues/40**
i had the same problem

Reply:
then we can remove `$this->router->getBaseUrl();` if we can get it from request?

Reply:
yes and no, to get it from the router is much better, as to use it from the request, should just be a fallback.